integration-tests/tests/src/tests/sync/flexible.ts

```diff
@@ -1480,6 +1483,8 @@ describe.skipIf(environment.missingServer, "Flexible sync", function () {
         mutableSubs.add(newRealm.objects(FlexiblePersonSchema.name).filtered("age > 30"));
       });
 
+      await newRealm?.syncSession?.uploadAllLocalChanges();
```

Comment:
This one does not make sense to me, as we have not created any local objects prior to calling it (except in `addPersonAndResyncWithSubscription`, which you have fixed) – we've only added a subscription, which should be handled by `await subs.update`.

Reply:
It fails without this :(